### PR TITLE
bugfix/1249 - command parser: discard empty tokens

### LINKED ISSUE #1249: Timewarp with 2 spaces breaks everything
Browser: Chrome

Airport: Any

#### Steps to reproduce
- Run `timewarp  5` (2 spaces between the command and the value)
- Observe everything break

### PR DESCRIPTION
**Pull request by cake-pie**
Resolves #1249

The purpose of this pull request is to make the command parser robust against typos of consecutive spaces. Previously, this would result in empty tokens when the command string is split using space as a delimiter. Some commands broke when empty strings were ingested in place of expected arguments. Such empty tokens are now discarded.

### CODE REVIEW

File: src/assets/scripts/client/commands/parsers/CommandParser.js [hunk 1]
```diff
@@ -128,7 +128,7 @@ export default class CommandParser {
      */
     _extractCommandsAndArgs(rawCommandWithArgs) {
         const commandOrCallsignIndex = 0;
-        const commandArgSegmentsWithCallsign = rawCommandWithArgs.split(COMMAND_ARGS_SEPARATOR);
+        const commandArgSegmentsWithCallsign = rawCommandWithArgs.split(COMMAND_ARGS_SEPARATOR).filter(t => t);
```
Comment (n8rzz) [inline src/assets/scripts/client/commands/parsers/CommandParser.js]: What exactly are we `.filter(t => t)` here? 

Oh, I think I see it. An empty string will evaluate as falsy here and will not be returned. Is that correct? If so, that's clever.

File: src/assets/scripts/client/commands/parsers/CommandParser.js [hunk 2]
```diff
@@ -128,7 +128,7 @@ export default class CommandParser {
      */
     _extractCommandsAndArgs(rawCommandWithArgs) {
         const commandOrCallsignIndex = 0;
-        const commandArgSegmentsWithCallsign = rawCommandWithArgs.split(COMMAND_ARGS_SEPARATOR);
+        const commandArgSegmentsWithCallsign = rawCommandWithArgs.split(COMMAND_ARGS_SEPARATOR).filter((t) => t);
```
Comment (n8rzz) [inline src/assets/scripts/client/commands/parsers/CommandParser.js:131]: Is this coercing an empty string (space) to falsey and filtering those out? I've never seen this before. If that's it, this is clever.

**Review by n8rzz — APPROVED** [review]
You're probably digging into this already, but I'd love to see some tests for this too. Especially since this is a bugfix. Otherwise looks good, clean and simple. Thank you for your contribution, we really appreciate it!

File: src/assets/scripts/client/commands/parsers/CommandParser.js (hunk 1 above)
Comment (cake-pie) [reply src/assets/scripts/client/commands/parsers/CommandParser.js]: Yes, that's pretty much it.

**Comment by cake-pie** [comment]
Added test, cleaned up commits, all set.

File: test/commands/parsers/CommandParser.spec.js [hunk 3]
```diff
@@ -76,6 +77,17 @@ ava('sets #commandList with AircraftCommandModel objects when it receives transm
     });
 });
 
+ava('._extractCommandsAndArgs() discards empty tokens caused by multiple spaces', t => {
```
Comment (n8rzz) [inline test/commands/parsers/CommandParser.spec.js:99]: Love it!

**Review by n8rzz — COMMENTED** [review]
Thanks for adding that test!

**Pull request merged**